responsibleai/responsibleai/rai_insights/rai_insights.py

```diff
@@ -643,8 +621,8 @@ def _validate_rai_insights_input_parameters(
             # Ensure that the model has the required methods and that they
             # do not change the input data.
             if task_type != ModelTask.FORECASTING:
-                self._ensure_model_outputs(input_data=small_train_data)
-            self._ensure_model_outputs(input_data=small_test_data)
+                self._ensure_model_outputs(input_data=actual_train_data)
```

Comment:
In some cases calling predict can be expensive on some models - by using all data we can really slow down our dashboard computations much more.  Perhaps we will need to look into some caching logic for such scenarios, since we seem to be calling predict on all data many times but not re-using the results anywhere.

Reply:
I agree we seem to be doing quite a few prediction calls. Caching these predictions maybe is out of scope of this PR. Maybe I can follow up on this later.